orchagent/observer.h

```diff
@@ -30,26 +38,64 @@ class Observer
 class Subject
```

Comment:
>Subject [](http://example.com/codeflow?start=6&length=7)

The Subject class stands here for long and widely used. Please create a new class if really necessary, you may reuse code by inheritance or composition, or even by refactorying existing class implementation. Do not introduce disruptive change in comon code.

This chagne itself could be a standalone PR. Do not mix with business logic.

Reply:
@prsunny , @qiluo-msft : These changes are enhancement to support registering for specific message types. I have removed the code for now reverted the old changes. As suggested we may push these part a separate PR.
Thanks,
-Praveen.